runtime/compiler/env/J9ClassEnv.cpp

```diff
@@ -220,6 +222,60 @@ J9::ClassEnv::isAnonymousClass(TR::Compilation *comp, TR_OpaqueClassBlock *clazz
    return comp->fej9()->isAnonymousClass(clazz);
    }
 
+const TR::TypeLayout*
+J9::ClassEnv::enumerateFields(TR::Region& region, TR_OpaqueClassBlock * opaqueClazz, TR::Compilation *comp)
+   {  
+   J9Class *clazz = (J9Class*)opaqueClazz;
+   TR_VMFieldsInfo fieldsInfo(comp, clazz, 1, stackAlloc);
+   ListIterator<TR_VMField> iter(fieldsInfo.getFields());
+   TR::TypeLayoutBuilder tlb(region);
+   for (TR_VMField *field = iter.getFirst(); field; field = iter.getNext())
+      {
+      char *signature = field->signature;
+      char charSignature = *signature;
+      TR::DataType dataType;
+      switch(charSignature)
+         {
+         case 'Z': 
+         case 'B': 
+         case 'C': 
+         case 'S': 
+         case 'I':
+            {
+            dataType = TR::Int32;
+            break;
+            }
+         case 'J':
+            {
+            dataType = TR::Int64;
+            break;
+            }
+         case 'F':
+            {
+            dataType = TR::Float;
+            break;
+            }
+         case 'D':
+            {
+            dataType = TR::Double;
+            break;
+            }
+         case 'L': 
+         case '[':
+            {
+            dataType = TR::Address;
+            break;
+            }
+         }
+      size_t nameSize = strlen(field->name)+1;
+      char *fieldName = new (region) char[nameSize];
+      strncpy(fieldName, field->name, nameSize);
+      TR_ASSERT_FATAL(fieldName[nameSize-1] == '\0', "fieldName buffer was too small.");
+      int32_t offset = field->offset + TR::Compiler->om.objectHeaderSizeInBytes();
+      tlb.add(TR::TypeLayoutEntry(dataType, offset, fieldName));
```

Comment:
To aid debugging I think this could use a guarded traceMsg to describe the fields and their offsets that the builder is finding.

Reply:
The function itself is a util function, so there are two options for me to add the guard. I can either choose to pass in a boolean from the caller or add an OMR option indicating whether traceMsg is needed. Could you give me some suggestions?

Reply:
hmm true that would make it a bit messy - I'll withdraw the suggestion since I had forgotten where the code was - we should add trace to where this is called from to report the result - eg once we have built a type layout or whatever, but that can be a separate PR.